dev/io.openliberty.http.monitor/src/io/openliberty/http/monitor/ServletFilter.java

```diff
@@ -322,46 +320,88 @@ private String resolveHttpRoute(SRTServletRequest srtServletRequest, String cont
 					}
 				}
 				
+				/*
+				 * Explicitly deal with resources loaded by JSF / Jakarta Faces
+				 * 
+				 * URL for resources request will end with the file extension of the original requesting 
+				 * 
+				 * i.e., page.xhtml -> /jakarta.faces.resource/someFile.file.xhtml
+				 */
+				else if (servletPath.startsWith("/jakarta.faces.resource") || servletPath.startsWith("/javax.faces.resource")) {
+					String[] arr = servletPath.split("\\.");
+					String extension = arr[arr.length-1];
+					httpRoute = contextPath + "/*." + extension;
+				}
+
 				else {
 					/*
-					 * PATH INFO NULL
-					 * SERVLET NOT NULL
+					 * PATH INFO = NULL
+					 * SERVLET PATH = NOT NULL
+					 * 
 					 * We are dealing with a direct match.
 					 * OR a wild card servlet , but right on the node
 					 * ^ Can only resolve for EE8
 					 * e.g.
-					 * servlet path: /path/*
-					 * request path: /path
+					 * servlet path is specified as : /path/*
+					 * request path was requested as : /path
 					 *
 					 */
 
-					//Used by EE8 and above
+					/*
+					 *  Applies to Servlet 4 and up (EE8 and up ) as we are using the 'pattern' value
+					 *  that is only obtainable with WebAppDispatcherContext40.
+					 */
 					if (Servlet4Helper.isServlet4Up()) {
 
 						String pattern = Servlet4Helper.getPattern(iwadc);
-						//This resolves the /wild/* servlet path and the /wild is entered for EE8
-						if (pattern != null && pattern.endsWith("/*")) {
-							httpRoute = contextPath + pattern;
+						
+						/*
+						 * This resolves the /wild/* servlet path and the /wild is entered for EE8
+						 * Also if configured with *.<extension> (i.e., *.xhtml, *.jsf, *.abc)
+						 */
+						if (pattern != null && (pattern.endsWith("/*") || pattern.startsWith("*."))) {
+							httpRoute = contextPath + "/" + pattern;
 							//direct mtach for EE8
 						} else {
 							httpRoute = contextPath + servletPath;
 						}
 					}
-					//EE7 specifically
+					/*
+					 * Applies only to EE7 
+					 */
 					else {
+						
 						/*
-						 * Simply a direct match.
+						 * Hard coded to deal with the typical file extensions of JSF files : `.jsf`, `.faces` or `.xhtml`. This
+						 * is will catch.
 						 * 
-						 * Unfortunately, given a scenario where a servlet is set as /path/*
-						 * we cannot properly resolve the http route of hits to /path as /path/*
+						 * Unfortunately, won't be able to catch any servlet mappings where they mapped the JSF servlet to some
+						 * random extension (i.e., *.abc)
 						 * 
-						 * Constraint of EE7 where we have no "pattern" information
+						 * In those cases, it'll just be handled as a direct match (i.e., the bottom else block)
 						 */
-						httpRoute = contextPath + servletPath;
-						
+						if (servletPath.endsWith(".jsf")) {
```

Comment:
Will there be upper case of jsf, faces or xhtml?

Reply:
The proper way would be lower case. upper case file extension would be something different.
@volosied  For JEE7, we're not able to resolve the `pattern` value from the servlet filter so we need to do direct matching.
Are capital `.JSF` and `.FACES` and `.XHTML` _"valid"_ values? Or should is it only lower case.

Reply:
Servlet mappings are case sensitive.

Typical jsf mappings are `.jsf`, `.faces`, and `.xhtml`, and `/faces/*`. 

However, they are customizable via the servlet mapping in the web.xml, so some customers could have other settings.

Reply:
Yup, since we don't have mapping/pattern information in EE7, we can only anticipate the commonly/expected use of file extensions to map to the JSF servlet (i.e., the `.jsf` and `.faces` and .xhtml, `/faces/*` *is something* that we can catch in EE7). That being said, capital JSF and FACES and XHTML are not the usual use case?

@fmhwong